### PR TITLE
FAD-6478 Modal improvements

### PR DESCRIPTION
**Pull request by jonambas**
Matchbox PR: https://github.com/SparkPost/matchbox/pull/120

Intentionally left out the onClose prop in of some of the TFA modals

### CODE REVIEW

File: src/components/modals/tests/Modal.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  });
+
+  it('should not render contents when not open', () => {
+    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find('Modal').dive().find('TransitionGroup').children()).toHaveLength(0);
+  });
+
+  it('should render contents when not open', () => {
+    wrapper.setProps({ open: true });
+    expect(wrapper.find('Modal').dive().find('TransitionGroup').children()).toMatchSnapshot();
```
Comment (jonambas) [inline src/components/modals/tests/Modal.test.js]: This test will fail until matchbox version is updated

File: src/components/modals/tests/Modal.test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  });
+
+  it('should not render contents when not open', () => {
+    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find('Modal').dive().find('TransitionGroup').children()).toHaveLength(0);
+  });
+
+  it('should render contents when open', () => {
+    wrapper.setProps({ open: true });
+    expect(wrapper.find('Modal').dive().find('TransitionGroup').children()).toMatchSnapshot();
```
Comment (jonambas) [inline src/components/modals/tests/Modal.test.js]: Actually, I should probably remove these tests / add them to matchbox tests when those exist

File: src/pages/billing/components/BillingSummary.js [hunk 3]
```diff
@@ -27,12 +27,6 @@ export default class BillingSummary extends Component {
   handleContactModal = () => this.handleModal(CONTACT_MODAL);
   handleIpModal = () => this.handleModal(IP_MODAL);
 
-  handleEscape = (e) => {
-    if (e.key === 'Escape') {
-      this.handleModal();
-    }
-  }
```
Comment (bkemper) [inline src/pages/billing/components/BillingSummary.js]: 👍

File: src/pages/reports/components/ShareModal.js [hunk 4]
```diff
@@ -91,7 +91,7 @@ export class ShareModal extends Component {
     return (
       <Fragment>
         <Button id='shareModalButton' disabled={this.props.disabled} fullWidth onClick={this.toggleModal}>Share</Button>
-        <Modal open={open}>
+        <Modal open={open} onClose={this.toggleModal}>
           {open && <WindowEvent event='keydown' handler={this.handleKeydown} />}
```
Comment (bkemper) [inline src/pages/reports/components/ShareModal.js]: Can't we remove this `WindowEvent` and `this.handleKeydown`?

Comment (jonambas) [reply src/pages/reports/components/ShareModal.js]: Not completely, this `handleKeydown` also works with `enter`, which matchbox `onClose` won't check. We can remove `onEscape(this.toggleModal)(e);` though

**Pull request merged**